frontend/lib/resolution/Resolver.cpp

```diff
@@ -1502,21 +1502,54 @@ void Resolver::issueErrorForFailedModuleDot(const Dot* dot,
 
 }
 
-void Resolver::handleResolvedCall(ResolvedExpression& r,
-                                  const uast::AstNode* astForErr,
-                                  const CallInfo& ci,
-                                  const CallResolutionResult& c) {
+bool Resolver::handleResolvedCallWithoutError(ResolvedExpression& r,
+                                              const uast::AstNode* astForErr,
+                                              const CallInfo& ci,
+                                              const CallResolutionResult& c) {
 
   if (!c.exprType().hasTypePtr()) {
-    issueErrorForFailedCallResolution(astForErr, ci, c);
     r.setType(QualifiedType(r.type().kind(), ErroneousType::get(context)));
+    return true;
   } else {
     r.setMostSpecific(c.mostSpecific());
     r.setPoiScope(c.poiInfo().poiScope());
     r.setType(c.exprType());
     // gather the poi scopes used when resolving the call
     poiInfo.accumulate(c.poiInfo());
   }
+  return false;
+}
+
+void Resolver::handleResolvedCall(ResolvedExpression& r,
+                                  const uast::AstNode* astForErr,
+                                  const CallInfo& ci,
+                                  const CallResolutionResult& c) {
+
+  if (handleResolvedCallWithoutError(r, astForErr, ci, c)) {
+    issueErrorForFailedCallResolution(astForErr, ci, c);
+  }
+}
+
+void Resolver::handleResolvedCallPrintCandidates(ResolvedExpression& r,
+                                                 const uast::Call* call,
+                                                 const CallInfo& ci,
+                                                 const Scope* scope,
+                                                 const PoiScope* poiScope,
+                                                 const QualifiedType& receiverType,
+                                                 const CallResolutionResult& c) {
+
+  if (handleResolvedCallWithoutError(r, call, ci, c)) {
+    std::vector<ApplicabilityResult> rejected;
+    std::ignore = resolveCallInMethod(context, call, ci, scope, poiScope,
+                                      receiverType, &rejected);
+
+    if (rejected.size() == 0) {
+      issueErrorForFailedCallResolution(call, ci, c);
+      return;
+    }
+
+    CHPL_REPORT(context, NoMatchingCandidates, call, ci, rejected);
```

Comment:
I've made it call `NoMatchingCandidates` (with an empty vector, since there is not enough information to re-compute the list of excluded candidates). I'll leave computing the list as another future work item if possible.